flows/definition/legacy.go

```diff
@@ -93,12 +94,40 @@ func (c *legacyContactReference) Migrate() *flows.ContactReference {
 }
 
 type legacyGroupReference struct {
-	UUID flows.GroupUUID `json:"uuid"`
-	Name string          `json:"name"`
+	UUID flows.GroupUUID
+	Name string
+}
+
+func (g *legacyGroupReference) Migrate() *flows.GroupReference {
+	return flows.NewGroupReference(g.UUID, g.Name)
 }
 
-func (g *legacyGroupReference) Migrate() *flows.Group {
-	return flows.NewGroup(g.UUID, g.Name)
+func (g *legacyGroupReference) UnmarshalJSON(data []byte) error {
+	// group reference may be a string
+	if data[0] == '"' {
```

Comment:
Not sure if there's a cleaner way to do this but this handles the case where the group is just a string instead of a dictionary.